crates/uv-cli/src/lib.rs

```diff
@@ -1962,53 +1980,98 @@ pub struct InitArgs {
 #[derive(Args)]
 #[allow(clippy::struct_excessive_bools)]
 pub struct RunArgs {
-    /// Include optional dependencies from the extra group name; may be provided more than once.
+    /// Include optional dependencies from the extra group name.
     ///
-    /// Only applies to `pyproject.toml`, `setup.py`, and `setup.cfg` sources.
+    /// May be provided more than once.
+    ///
+    /// Optional dependencies are defined at `project.optional-dependencies` in
+    /// a `pyproject.toml`.
+    ///
+    /// This option is only available when running in a project or invoking a
+    /// script with inline metadata.
     #[arg(long, conflicts_with = "all_extras", value_parser = extra_name_with_clap_error)]
     pub extra: Option<Vec<ExtraName>>,
 
     /// Include all optional dependencies.
     ///
-    /// Only applies to `pyproject.toml`, `setup.py`, and `setup.cfg` sources.
+    /// Optional dependencies are defined at `project.optional-dependencies` in
+    /// a `pyproject.toml`.
+    ///
+    /// This option is only available when running in a project or invoking a
+    /// script with inline metadata.
     #[arg(long, conflicts_with = "extra")]
     pub all_extras: bool,
 
     #[arg(long, overrides_with("all_extras"), hide = true)]
     pub no_all_extras: bool,
 
     /// Include development dependencies.
+    ///
+    /// Development dependencies are defined at `tool.uv.dev-dependencies` in a
+    /// `pyproject.toml`.
+    ///
+    /// This option is only available when running in a project or invoking a
+    /// script with inline metadata.
```

Comment:
I don't think this applies to inline metadata, does it?

Reply:
I'm not sure. You can have a `tool.uv` section? I guess I'll check

Reply:
I don't _think_ we would respect that, because we return:

```rust
/// PEP 723 metadata as parsed from a `script` comment block.
///
/// See: <https://peps.python.org/pep-0723/>
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "kebab-case")]
pub struct Pep723Metadata {
    pub dependencies: Vec<pep508_rs::Requirement<VerbatimParsedUrl>>,
    pub requires_python: Option<pep440_rs::VersionSpecifiers>,
}
```

Reply:
Yeah I confirmed we do not, which is debatably wrong. I'm fine with not reading `[tool.uv.dev-dependencies]` but we do need to respect `[tool.uv]` settings: #5855